prow/genjobs/cmd/genjobs/main.go

```diff
@@ -50,66 +51,70 @@ type options struct {
 	jobBlacklist     sets.String
+	repoWhitelist    sets.String
+	repoBlacklist    sets.String
 	jobType          sets.String
 	clean            bool
 	dryRun           bool
 	extraRefs        bool
+	resolve          bool
 	sshClone         bool
 	overrideSelector bool
 }
 
 // parseFlags parses the command-line flags.
 func (o *options) parseFlags() {
 	var (
-		_repoWhitelist []string
-		_repoBlacklist []string
 		_jobWhitelist  []string
 		_jobBlacklist  []string
+		_repoWhitelist []string
+		_repoBlacklist []string
 		_jobType       []string
 	)
 
-	flag.StringVar(&o.bucket, "bucket", "private-build", "GCS bucket name to upload logs and build artifacts to.")
-	flag.StringSliceVar(&o.branches, "branches", []string{}, "Branch(es) to generate job(s) for.")
+	flag.StringVar(&o.bucket, "bucket", "", "GCS bucket name to upload logs and build artifacts to.")
+	flag.StringVar(&o.cluster, "cluster", "", "GCP cluster to run the job(s) in.")
 	flag.StringVar(&o.channel, "channel", "", "Slack channel to report job status notifications to.")
-	flag.StringToStringVar(&o.selector, "selector", map[string]string{}, "Node selector(s) to constrain job(s).")
-	flag.StringVar(&o.cluster, "cluster", "private", "GCP cluster to run the job(s) in.")
-	flag.BoolVar(&o.clean, "clean", false, "Clean output directory before job(s) generation.")
-	flag.BoolVar(&o.dryRun, "dry-run", false, "Run in dry run mode.")
-	flag.BoolVar(&o.overrideSelector, "override-selector", false, "The existing node selector will be overridden rather than added to.")
-	flag.BoolVar(&o.sshClone, "ssh-clone", false, "Enable a clone of the git repository over ssh.")
-	flag.BoolVar(&o.extraRefs, "extra-refs", false, "Apply translation to all extra refs regardless of mapping.")
-	flag.StringVar(&o.sshKeySecret, "ssh-key-secret", "ssh-key-secret", "GKE cluster secrets containing the Github ssh private key.")
+	flag.StringVar(&o.sshKeySecret, "ssh-key-secret", "", "GKE cluster secrets containing the Github ssh private key.")
 	flag.StringVar(&o.modifier, "modifier", defaultModifier, "Modifier to apply to generated file and job name(s).")
-	flag.StringToStringVarP(&o.labels, "labels", "l", map[string]string{}, "Prow labels to apply to the job(s).")
-	flag.StringToStringVarP(&o.env, "env", "e", map[string]string{}, "Environment variables to set for the job(s).")
 	flag.StringVarP(&o.input, "input", "i", ".", "Input directory containing job(s) to convert.")
 	flag.StringVarP(&o.output, "output", "o", ".", "Output directory to write generated job(s).")
-	flag.StringSliceVarP(&_repoWhitelist, "repo-whitelist", "w", []string{}, "Repositories to whitelist in generation process.")
-	flag.StringSliceVarP(&_repoBlacklist, "repo-blacklist", "b", []string{}, "Repositories to blacklist in generation process.")
+	flag.StringSliceVar(&o.branches, "branches", []string{}, "Branch(es) to generate job(s) for.")
+	flag.StringSliceVarP(&o.presets, "presets", "p", []string{}, "Path to file(s) containing additional presets.")
```

Comment:
"files or directories" is important to note since this should always point to `config.yaml` and the job config directory.

Reply:
For `--presets`, this is an arbitrary list of file(s). We explicitly perform an [extension check](https://github.com/istio/test-infra/pull/2099/files#diff-590584282c8f4a9811ba7e4ed678ccebR137-R140) to ensure these are files. This is also not limited to `config.yaml`. A use can add presets to the generated jobs from another file (in another directory or repo) as long as it passes [.ReadJobConfig(path)](https://github.com/istio/test-infra/pull/2099/files#diff-590584282c8f4a9811ba7e4ed678ccebR215) and has a `yaml` extension.